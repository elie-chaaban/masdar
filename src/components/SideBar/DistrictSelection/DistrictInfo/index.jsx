import React from 'react';
import PropTypes from 'prop-types';
import {useDispatch, useSelector} from 'react-redux';
import {
  selectDistrict,
  toggleInsight,
  toggleSlider,
  setIFrames,
  toggleInsightSlider,
  setIsIntersectionDataSliderOpened,
  toggleActionSlider
} from '../../../../reduxStore/actions';

const DistrictInfo = ({district, onClick, active}) => {
  const {name, location} = district.info.info;
  const dispatch = useDispatch();
  const {showSlider, insight, isOpenInsightSlider, isIntersectionDataSliderOpened, actionSlider} = useSelector((s) => s.dashboard);
  const changeDistrict = () => {
    if (!active) {
      onClick();
      // i need to close all iFrames and insights
      if (showSlider) dispatch(toggleSlider());
      if (insight) dispatch(toggleInsight(null));
      if (isOpenInsightSlider) dispatch(toggleInsightSlider());
      if (isIntersectionDataSliderOpened) dispatch(setIsIntersectionDataSliderOpened(false));
      if (actionSlider.open) dispatch(toggleActionSlider());
      dispatch(selectDistrict(district.info));
      dispatch(setIFrames(district.info.iFrames));
    }
  };
  return (
    <div className="district-card m-4" onClick={changeDistrict}>
      <div className="row px-4">
        <div className="col-7 p-0">
          <div className={active ? 'active' : ''}>
            <img className="img-fluid" src={district.image} alt="district" />
          </div>
          <div className="desc">
            <h5>{name}</h5>
            <p>{location}</p>
          </div>
        </div>
        <div className="col-5 p-0 pl-1 d-flex justify-content-between  flex-column">
          <div className="insights-info">
            <p>
              <span>{district.indexes.energy}</span> <small>Sustainability</small>
            </p>
          </div>
          <div className="insights-info">
            <p>
              <span>{district.indexes.security}</span> <small>Security</small>
            </p>
          </div>
          <div className="insights-info">
            <p>
              <span>{district.indexes.mobility}</span> <small>Mobility</small>
            </p>
          </div>
          <div className="insights-info">
            <p>
              <span>{district.indexes.facilities}</span> <small>Economy</small>
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};

DistrictInfo.propTypes = {
  district: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired
};
export default React.memo(DistrictInfo);
