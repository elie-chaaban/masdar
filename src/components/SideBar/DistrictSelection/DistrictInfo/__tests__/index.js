import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {mockStore} from '../../../../../testUtils';
import DistrictInfo from '..';

const data = {
  info: {
    info: {
      id: '1',
      name: 'Telus HQ',
      location: 'Edmonton, Canada',
      extrusion: ['59', '88', '79', '78', '80', '32', '8', '2', '4', '22', '3', '5', '60'],
      latitude: 53.54310552517575,
      longitude: -113.49581660415623,
      zoom: 15.721944548645808
    }
  },
  image: 'url',
  indexes: {mobility: 74, security: 62, facilities: 23, energy: 23}
};
const store = mockStore({});
const click = jest.fn();
test('District Info not selected ', async () => {
  const dom = mount(
    <Provider store={store}>
      <DistrictInfo district={data} onClick={click} active={false} />
    </Provider>
  );
  expect(dom).toMatchSnapshot();
  expect(dom.find('.district-card')).toHaveLength(1);
  expect(dom.find('.active')).toHaveLength(0);
  expect(dom.find('img').prop('src')).toBe('url');
  expect(dom.find('.desc').html()).toEqual('<div class="desc"><h5>Telus HQ</h5><p>Edmonton, Canada</p></div>');
  expect(dom.find('.insights-info').at(0).find('span').text()).toBe('23');
  expect(dom.find('.insights-info').at(1).find('span').text()).toBe('23');
  expect(dom.find('.insights-info').at(2).find('span').text()).toBe('62');
  expect(dom.find('.insights-info').at(3).find('span').text()).toBe('74');
  dom.find('.district-card').simulate('click');
  expect(click).toBeCalled();
  jest.clearAllMocks();
  dom.unmount();
});

test('District Info selected ', () => {
  const dom = mount(
    <Provider store={store}>
      <DistrictInfo district={data} onClick={click} active={true} />
    </Provider>
  );
  expect(dom).toMatchSnapshot();
  expect(dom.find('.active')).toHaveLength(1);
  dom.find('.district-card').simulate('click');
  expect(click).toBeCalledTimes(0);
  jest.clearAllMocks();
  dom.unmount();
});
