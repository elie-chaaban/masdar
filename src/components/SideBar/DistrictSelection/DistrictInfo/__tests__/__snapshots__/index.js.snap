// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`District Info not selected  1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo(DistrictInfo)
    active={false}
    district={
      Object {
        "image": "url",
        "indexes": Object {
          "energy": 23,
          "facilities": 23,
          "mobility": 74,
          "security": 62,
        },
        "info": Object {
          "info": Object {
            "extrusion": Array [
              "59",
              "88",
              "79",
              "78",
              "80",
              "32",
              "8",
              "2",
              "4",
              "22",
              "3",
              "5",
              "60",
            ],
            "id": "1",
            "latitude": 53.54310552517575,
            "location": "Edmonton, Canada",
            "longitude": -113.49581660415623,
            "name": "Telus HQ",
            "zoom": 15.721944548645808,
          },
        },
      }
    }
    onClick={[MockFunction]}
  >
    <div
      className="district-card m-4"
      onClick={[Function]}
    >
      <div
        className="row px-4"
      >
        <div
          className="col-7 p-0"
        >
          <div
            className=""
          >
            <img
              alt="district"
              className="img-fluid"
              src="url"
            />
          </div>
          <div
            className="desc"
          >
            <h5>
              Telus HQ
            </h5>
            <p>
              Edmonton, Canada
            </p>
          </div>
        </div>
        <div
          className="col-5 p-0 pl-1 d-flex justify-content-between  flex-column"
        >
          <div
            className="insights-info"
          >
            <p>
              <span>
                23
              </span>
               
              <small>
                Energy
              </small>
            </p>
          </div>
          <div
            className="insights-info"
          >
            <p>
              <span>
                23
              </span>
               
              <small>
                Economy
              </small>
            </p>
          </div>
          <div
            className="insights-info"
          >
            <p>
              <span>
                62
              </span>
               
              <small>
                Security
              </small>
            </p>
          </div>
          <div
            className="insights-info"
          >
            <p>
              <span>
                74
              </span>
               
              <small>
                Mobility
              </small>
            </p>
          </div>
        </div>
      </div>
    </div>
  </Memo(DistrictInfo)>
</Provider>
`;

exports[`District Info selected  1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo(DistrictInfo)
    active={true}
    district={
      Object {
        "image": "url",
        "indexes": Object {
          "energy": 23,
          "facilities": 23,
          "mobility": 74,
          "security": 62,
        },
        "info": Object {
          "info": Object {
            "extrusion": Array [
              "59",
              "88",
              "79",
              "78",
              "80",
              "32",
              "8",
              "2",
              "4",
              "22",
              "3",
              "5",
              "60",
            ],
            "id": "1",
            "latitude": 53.54310552517575,
            "location": "Edmonton, Canada",
            "longitude": -113.49581660415623,
            "name": "Telus HQ",
            "zoom": 15.721944548645808,
          },
        },
      }
    }
    onClick={[MockFunction]}
  >
    <div
      className="district-card m-4"
      onClick={[Function]}
    >
      <div
        className="row px-4"
      >
        <div
          className="col-7 p-0"
        >
          <div
            className="active"
          >
            <img
              alt="district"
              className="img-fluid"
              src="url"
            />
          </div>
          <div
            className="desc"
          >
            <h5>
              Telus HQ
            </h5>
            <p>
              Edmonton, Canada
            </p>
          </div>
        </div>
        <div
          className="col-5 p-0 pl-1 d-flex justify-content-between  flex-column"
        >
          <div
            className="insights-info"
          >
            <p>
              <span>
                23
              </span>
               
              <small>
                Energy
              </small>
            </p>
          </div>
          <div
            className="insights-info"
          >
            <p>
              <span>
                23
              </span>
               
              <small>
                Economy
              </small>
            </p>
          </div>
          <div
            className="insights-info"
          >
            <p>
              <span>
                62
              </span>
               
              <small>
                Security
              </small>
            </p>
          </div>
          <div
            className="insights-info"
          >
            <p>
              <span>
                74
              </span>
               
              <small>
                Mobility
              </small>
            </p>
          </div>
        </div>
      </div>
    </div>
  </Memo(DistrictInfo)>
</Provider>
`;
